backend/app/controllers/api/v1/sessions_controller.rb

```diff
@@ -38,6 +45,13 @@ def refresh
 
       private
 
+      attr_reader :headers
```

Comment:
You don't need to extract this into instance variable, because you're using it only in one method. You could just directly use
```ruby
service.new.call(request.headers)
```